feat(linter/jsx-a11y): add fixer for anchor-has-content

DonIsaac opened the pull request:
Add a conditional fix that removes `aria-hidden` from an anchor's child if there
is only a single child. This PR also fixes a false positive on hidden anchors.
It should report visible anchors with hidden content, not hidden anchors.

DonIsaac commented:
* **#4852** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/4852?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @DonIsaac and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="11px" height="11px"/> <b>Graphite</b></a>
  <!-- Current dependencies on/for this PR: -->

camc314 commented on crates/oxc_linter/src/rules/jsx_a11y/anchor_has_content.rs:
```diff
@@ -70,7 +69,9 @@ impl Rule for AnchorHasContent {
             if name == "a" {
                 if is_hidden_from_screen_reader(ctx, &jsx_el.opening_element) {
-                    ctx.diagnostic(remove_aria_hidden(jsx_el.span));
+                    // ctx.diagnostic_with_suggestion(remove_aria_hidden(jsx_el.span), |_fixer| {
+                    //     remove_hidden_attributes(&jsx_el.opening_element)
+                    // });
```
remove/ uncomment?

camc314 commented on crates/oxc_linter/src/rules/jsx_a11y/anchor_has_content.rs:
```diff
@@ -114,12 +146,27 @@ fn test() {
+        (r"<a aria-hidden><span aria-hidden>Foo</span></a>", None, None),
+        (r#"<a hidden="true">Foo</a>"#, None, None),
+        (r#"<a hidden="">Foo</a>"#, None, None),
+        // TODO: should these be failing?
+        (r"<a><Bar hidden /></a>", None, None),
+        (r#"<a><Bar hidden="" /></a>"#, None, None),
+        (r#"<a><Bar hidden="until-hidden" /></a>"#, None, None),
```
good question 🙂 probably not? if they are capitalized, that infers that it's a react component rather than a html one. so should be allowed?

DonIsaac replied:
> good question 🙂 probably not? if they are capitalized, that infers that it's a react component rather than a html one. so should be allowed?

It has less to do with `Bar` and more to do with `hidden`. `aria-hidden` gets reported, but `hidden` does not.

DonIsaac committed:
feat(linter/jsx-a11y): add fixer for anchor-has-content (#4852)

Add a conditional fix that removes `aria-hidden` from an anchor's child if there
is only a single child. This PR also fixes a false positive on hidden anchors.
It should report visible anchors with hidden content, not hidden anchors.
